app/controllers/users_controller.rb

```diff
@@ -5,12 +5,17 @@ def update
     flash.clear
     @user = User.find_by(id: session[:current_user_id])
     sid = params[:user][:sid]
+    email = params[:user][:email]
     if sid.blank? or sid.length != 10
-      redirect_to login_confirm_path, flash: { :error => "Invalid Student ID Number" }
+      redirect_to login_confirm_path, flash: { :error => "Invalid Student ID Number." }
+      return
+    end
+    if email.blank? or not email.match(/.+(@berkeley.edu)/)
+      redirect_to login_confirm_path, flash: { :error => "Please enter you berkeley email." }
```

Comment:
Should the flash message be "Please use your berkeley email to log in?"

Reply:
fixed!